From c3444cddfa8fe807629b9fa7f019531460870875 Mon Sep 17 00:00:00 2001
Message-ID: <c3444cddfa8fe807629b9fa7f019531460870875.1713654571.git.sharlatanus@gmail.com>
In-Reply-To: <cover.1713654571.git.sharlatanus@gmail.com>
References: <cover.1713654571.git.sharlatanus@gmail.com>
From: Sharlatan Hellseher <sharlatanus@gmail.com>
Date: Wed, 17 Apr 2024 22:24:02 +0100
Subject: [PATCH 35/47] gnu: python-regions: Update to 0.9.

* gnu/packages/astronomy.scm (python-regions): Update to 0.9.
[arguments] <#:phases>: Add 'create-setup.py phase, rename
'prepare-test-environment to 'build-extensions to reflect purpose.
[native-inputs]: Remove python-cython; add python-cython-3.

Change-Id: Ia634f93336eb7658c4eaeeb7ddb6539c2e0adee3
---
 gnu/packages/astronomy.scm | 21 ++++++++++++++-------
 1 file changed, 14 insertions(+), 7 deletions(-)

diff --git a/gnu/packages/astronomy.scm b/gnu/packages/astronomy.scm
index 10eb640e56..fe1bc54157 100644
--- a/gnu/packages/astronomy.scm
+++ b/gnu/packages/astronomy.scm
@@ -2667,13 +2667,13 @@ (define-public python-pyvo
 (define-public python-regions
   (package
     (name "python-regions")
-    (version "0.8")
+    (version "0.9")
     (source
      (origin
        (method url-fetch)
        (uri (pypi-uri "regions" version))
        (sha256
-        (base32 "09401pz7926zlci7cznd78hmv9947f6jxyy2afqdqc1xaccpzcq2"))))
+        (base32 "0kvfdzqry3vcvphd7ldmppbgn3ab97hbfzwxfrlxls92yi41h3i6"))))
     (build-system pyproject-build-system)
     (arguments
      (list
@@ -2681,14 +2681,21 @@ (define-public python-regions
       #~(list "-n" "auto")
       #:phases
       #~(modify-phases %standard-phases
+          ;; setup.py was removed in 84c80a280431adda00641cda5264c7de18b43b2f
+          ;; for some unknown reason, which caused the package to fail to
+          ;; build. It is being recreated based on that commit.
+          (add-after 'unpack 'create-setup.py
+            (lambda _
+              (call-with-output-file "setup.py"
+                (lambda (port)
+                  (format port "from setuptools import setup
+from extension_helpers import get_extensions
+setup(ext_modules=get_extensions())")))))
           ;; This file is opened in both install and check phases.
-          ;; XXX: Check if it is still required.
           (add-before 'install 'writable-compiler
             (lambda _ (make-file-writable "regions/_compiler.c")))
-          (add-before 'check 'prepare-test-environment
+          (add-before 'check 'build-extensions
             (lambda _
-              (setenv "HOME" "/tmp")
-              (make-file-writable "regions/_compiler.c")
               (invoke "python" "setup.py" "build_ext" "--inplace"))))))
     (propagated-inputs
      (list python-astropy
@@ -2698,7 +2705,7 @@ (define-public python-regions
            python-scipy
            python-shapely))
     (native-inputs
-     (list python-cython
+     (list python-cython-3
            python-extension-helpers
            python-pytest-arraydiff
            python-pytest-astropy
-- 
2.41.0

